### ISSUE TITLE
Crash when compiling

### ISSUE THREAD

**Issue by ammen99** [llvm/llvm-project#94662]
Hello everyone, after a system update compilation of my project (https://github.com/WayfireWM/wayfire) with clang and address/undefined sanitizers seems to fail and the error messages seem to indicate a bug in clang/llvm. I am attaching the output of the compiler plus the indicated log files, let me know if I can assist in some other way:

```
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: /usr/bin/clang++ -fdiagnostics-color=always -fsanitize=address,undefined -fno-omit-frame-pointer -Wall -Winvalid-pch -Wextra -std=c++17 -O2 -g -Wno-unused-parameter -fPIC -Iplugins/wm-actions/libwm-actions.so.p -Iplugins/wm-actions -I../plugins/wm-actions -I../src/api -I. -I.. -Iplugins/common -I../plugins/common -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/libevdev-1.0 -I/usr/include/libxml2 -D_GLIBCXX_ASSERTIONS=1 -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_EXTENSIVE -D_FILE_OFFSET_BITS=64 -DWAYFIRE_VERSION=\"0.9.0\" -DWLR_USE_UNSTABLE -DWF_USE_CONFIG_H -DWITH_GZFILEOP -c -MD -MQ plugins/wm-actions/libwm-actions.so.p/wm-actions.cpp.o -MF plugins/wm-actions/libwm-actions.so.p/wm-actions.cpp.o.d -fcolor-diagnostics -o plugins/wm-actions/libwm-actions.so.p/wm-actions.cpp.o ../plugins/wm-actions/wm-actions.cpp
1.	<eof> parser at end of file
2.	../src/api/wayfire/per-output-plugin.hpp:72:18: instantiating function definition 'wf::per_output_tracker_mixin_t<wayfire_wm_actions_output_t>::handle_new_output'
3.	/usr/bin/../lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/unique_ptr.h:1075:5: instantiating function definition 'std::make_unique<wayfire_wm_actions_output_t>'
4.	/usr/bin/../lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/std_function.h:435:2: instantiating function definition 'std::function<bool (const wf::activator_data_t &)>::function<wayfire_wm_actions_output_t::(lambda at ../plugins/wm-actions/wm-actions.cpp:314:46), void>'
5.	/usr/bin/../lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/std_function.h:288:7: instantiating function definition 'std::_Function_handler<bool (const wf::activator_data_t &), wayfire_wm_actions_output_t::(lambda at ../plugins/wm-actions/wm-actions.cpp:314:46)>::_M_invoke'
6.	/usr/bin/../lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/invoke.h:104:5: instantiating function definition 'std::__invoke_r<bool, wayfire_wm_actions_output_t::(lambda at ../plugins/wm-actions/wm-actions.cpp:314:46) &, const wf::activator_data_t &>'
7.	/usr/bin/../lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/invoke.h:60:5: instantiating function definition 'std::__invoke_impl<bool, wayfire_wm_actions_output_t::(lambda at ../plugins/wm-actions/wm-actions.cpp:314:46) &, const wf::activator_data_t &>'
8.	../plugins/wm-actions/wm-actions.cpp:314:46: instantiating function definition 'wayfire_wm_actions_output_t::(anonymous class)::operator()<wf::activator_data_t>'
 #0 0x00007fe96d00440a llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/lib64/libLLVM.so.18.1+0x60440a)
 #1 0x00007fe96d001bf4 llvm::sys::RunSignalHandlers() (/lib64/libLLVM.so.18.1+0x601bf4)
 #2 0x00007fe96cf360e4 (/lib64/libLLVM.so.18.1+0x5360e4)
 #3 0x00007fe96c453710 __restore_rt (/lib64/libc.so.6+0x40710)
 #4 0x00007fe975078f09 clang::InitializationSequence::InitializeFrom(clang::Sema&, clang::InitializedEntity const&, clang::InitializationKind const&, llvm::MutableArrayRef<clang::Expr*>, bool, bool) (/lib64/libclang-cpp.so.18.1+0x1478f09)
 #5 0x00007fe9750a1b6a clang::Sema::BuildCaptureInit(clang::sema::Capture const&, clang::SourceLocation, bool) (/lib64/libclang-cpp.so.18.1+0x14a1b6a)
 #6 0x00007fe9750a243c clang::Sema::BuildLambdaExpr(clang::SourceLocation, clang::SourceLocation, clang::sema::LambdaScopeInfo*) (/lib64/libclang-cpp.so.18.1+0x14a243c)
 #7 0x00007fe9753a4400 (/lib64/libclang-cpp.so.18.1+0x17a4400)
 #8 0x00007fe97539536e (/lib64/libclang-cpp.so.18.1+0x179536e)
 #9 0x00007fe97538b817 (/lib64/libclang-cpp.so.18.1+0x178b817)
#10 0x00007fe9753977c7 (/lib64/libclang-cpp.so.18.1+0x17977c7)
#11 0x00007fe9753b0245 (/lib64/libclang-cpp.so.18.1+0x17b0245)
#12 0x00007fe97539d2bd (/lib64/libclang-cpp.so.18.1+0x179d2bd)
#13 0x00007fe97538a118 clang::Sema::SubstStmt(clang::Stmt*, clang::MultiLevelTemplateArgumentList const&) (/lib64/libclang-cpp.so.18.1+0x178a118)
#14 0x00007fe9753d91c0 clang::Sema::InstantiateFunctionDefinition(clang::SourceLocation, clang::FunctionDecl*, bool, bool, bool) (/lib64/libclang-cpp.so.18.1+0x17d91c0)
#15 0x00007fe974eb98c7 clang::Sema::MarkFunctionReferenced(clang::SourceLocation, clang::FunctionDecl*, bool) (/lib64/libclang-cpp.so.18.1+0x12b98c7)
#16 0x00007fe974ebf797 (/lib64/libclang-cpp.so.18.1+0x12bf797)
#17 0x00007fe974e79e8e clang::Sema::MarkDeclRefReferenced(clang::DeclRefExpr*, clang::Expr const*) (/lib64/libclang-cpp.so.18.1+0x1279e8e)
#18 0x00007fe9751d776f (/lib64/libclang-cpp.so.18.1+0x15d776f)
#19 0x00007fe9751e0a8e clang::Sema::BuildCallToObjectOfClassType(clang::Scope*, clang::Expr*, clang::SourceLocation, llvm::MutableArrayRef<clang::Expr*>, clang::SourceLocation) (/lib64/libclang-cpp.so.18.1+0x15e0a8e)
#20 0x00007fe974e72d90 clang::Sema::BuildCallExpr(clang::Scope*, clang::Expr*, clang::SourceLocation, llvm::MutableArrayRef<clang::Expr*>, clang::SourceLocation, clang::Expr*, bool, bool) (/lib64/libclang-cpp.so.18.1+0x1272d90)
#21 0x00007fe974e915e3 clang::Sema::ActOnCallExpr(clang::Scope*, clang::Expr*, clang::SourceLocation, llvm::MutableArrayRef<clang::Expr*>, clang::SourceLocation, clang::Expr*) (/lib64/libclang-cpp.so.18.1+0x12915e3)
#22 0x00007fe975397921 (/lib64/libclang-cpp.so.18.1+0x1797921)
#23 0x00007fe9753b0245 (/lib64/libclang-cpp.so.18.1+0x17b0245)
#24 0x00007fe97539d2bd (/lib64/libclang-cpp.so.18.1+0x179d2bd)
#25 0x00007fe97538a118 clang::Sema::SubstStmt(clang::Stmt*, clang::MultiLevelTemplateArgumentList const&) (/lib64/libclang-cpp.so.18.1+0x178a118)
#26 0x00007fe9753d91c0 clang::Sema::InstantiateFunctionDefinition(clang::SourceLocation, clang::FunctionDecl*, bool, bool, bool) (/lib64/libclang-cpp.so.18.1+0x17d91c0)
#27 0x00007fe974eb98c7 clang::Sema::MarkFunctionReferenced(clang::SourceLocation, clang::FunctionDecl*, bool) (/lib64/libclang-cpp.so.18.1+0x12b98c7)
#28 0x00007fe974ebf797 (/lib64/libclang-cpp.so.18.1+0x12bf797)
#29 0x00007fe974e79e8e clang::Sema::MarkDeclRefReferenced(clang::DeclRefExpr*, clang::Expr const*) (/lib64/libclang-cpp.so.18.1+0x1279e8e)
#30 0x00007fe974e797fe clang::Sema::BuildDeclRefExpr(clang::ValueDecl*, clang::QualType, clang::ExprValueKind, clang::DeclarationNameInfo const&, clang::NestedNameSpecifierLoc, clang::NamedDecl*, clang::SourceLocation, clang::TemplateArgumentListInfo const*) (/lib64/libclang-cpp.so.18.1+0x12797fe)
#31 0x00007fe9751d41b4 clang::Sema::FixOverloadedFunctionReference(clang::Expr*, clang::DeclAccessPair, clang::FunctionDecl*) (/lib64/libclang-cpp.so.18.1+0x15d41b4)
#32 0x00007fe9751d5db6 (/lib64/libclang-cpp.so.18.1+0x15d5db6)
#33 0x00007fe9751d5b3a clang::Sema::BuildOverloadedCallExpr(clang::Scope*, clang::Expr*, clang::UnresolvedLookupExpr*, clang::SourceLocation, llvm::MutableArrayRef<clang::Expr*>, clang::SourceLocation, clang::Expr*, bool, bool) (/lib64/libclang-cpp.so.18.1+0x15d5b3a)
#34 0x00007fe974e732e5 clang::Sema::BuildCallExpr(clang::Scope*, clang::Expr*, clang::SourceLocation, llvm::MutableArrayRef<clang::Expr*>, clang::SourceLocation, clang::Expr*, bool, bool) (/lib64/libclang-cpp.so.18.1+0x12732e5)
#35 0x00007fe974e915e3 clang::Sema::ActOnCallExpr(clang::Scope*, clang::Expr*, clang::SourceLocation, llvm::MutableArrayRef<clang::Expr*>, clang::SourceLocation, clang::Expr*) (/lib64/libclang-cpp.so.18.1+0x12915e3)
#36 0x00007fe975397921 (/lib64/libclang-cpp.so.18.1+0x1797921)
#37 0x00007fe9753b0245 (/lib64/libclang-cpp.so.18.1+0x17b0245)
#38 0x00007fe9753b3844 (/lib64/libclang-cpp.so.18.1+0x17b3844)
#39 0x00007fe97539d2bd (/lib64/libclang-cpp.so.18.1+0x179d2bd)
#40 0x00007fe97538a118 clang::Sema::SubstStmt(clang::Stmt*, clang::MultiLevelTemplateArgumentList const&) (/lib64/libclang-cpp.so.18.1+0x178a118)
#41 0x00007fe9753d91c0 clang::Sema::InstantiateFunctionDefinition(clang::SourceLocation, clang::FunctionDecl*, bool, bool, bool) (/lib64/libclang-cpp.so.18.1+0x17d91c0)
#42 0x00007fe974eb98c7 clang::Sema::MarkFunctionReferenced(clang::SourceLocation, clang::FunctionDecl*, bool) (/lib64/libclang-cpp.so.18.1+0x12b98c7)
#43 0x00007fe974ebf797 (/lib64/libclang-cpp.so.18.1+0x12bf797)
#44 0x00007fe974e79e8e clang::Sema::MarkDeclRefReferenced(clang::DeclRefExpr*, clang::Expr const*) (/lib64/libclang-cpp.so.18.1+0x1279e8e)
#45 0x00007fe974e797fe clang::Sema::BuildDeclRefExpr(clang::ValueDecl*, clang::QualType, clang::ExprValueKind, clang::DeclarationNameInfo const&, clang::NestedNameSpecifierLoc, clang::NamedDecl*, clang::SourceLocation, clang::TemplateArgumentListInfo const*) (/lib64/libclang-cpp.so.18.1+0x12797fe)
#46 0x00007fe9751d41b4 clang::Sema::FixOverloadedFunctionReference(clang::Expr*, clang::DeclAccessPair, clang::FunctionDecl*) (/lib64/libclang-cpp.so.18.1+0x15d41b4)
#47 0x00007fe9751d5db6 (/lib64/libclang-cpp.so.18.1+0x15d5db6)
#48 0x00007fe9751d5b3a clang::Sema::BuildOverloadedCallExpr(clang::Scope*, clang::Expr*, clang::UnresolvedLookupExpr*, clang::SourceLocation, llvm::MutableArrayRef<clang::Expr*>, clang::SourceLocation, clang::Expr*, bool, bool) (/lib64/libclang-cpp.so.18.1+0x15d5b3a)
#49 0x00007fe974e732e5 clang::Sema::BuildCallExpr(clang::Scope*, clang::Expr*, clang::SourceLocation, llvm::MutableArrayRef<clang::Expr*>, clang::SourceLocation, clang::Expr*, bool, bool) (/lib64/libclang-cpp.so.18.1+0x12732e5)
#50 0x00007fe974e915e3 clang::Sema::ActOnCallExpr(clang::Scope*, clang::Expr*, clang::SourceLocation, llvm::MutableArrayRef<clang::Expr*>, clang::SourceLocation, clang::Expr*) (/lib64/libclang-cpp.so.18.1+0x12915e3)
#51 0x00007fe975397921 (/lib64/libclang-cpp.so.18.1+0x1797921)
#52 0x00007fe9753b0245 (/lib64/libclang-cpp.so.18.1+0x17b0245)
#53 0x00007fe97539d2bd (/lib64/libclang-cpp.so.18.1+0x179d2bd)
#54 0x00007fe97538a118 clang::Sema::SubstStmt(clang::Stmt*, clang::MultiLevelTemplateArgumentList const&) (/lib64/libclang-cpp.so.18.1+0x178a118)
#55 0x00007fe9753d91c0 clang::Sema::InstantiateFunctionDefinition(clang::SourceLocation, clang::FunctionDecl*, bool, bool, bool) (/lib64/libclang-cpp.so.18.1+0x17d91c0)
#56 0x00007fe9753dc107 clang::Sema::PerformPendingInstantiations(bool) (/lib64/libclang-cpp.so.18.1+0x17dc107)
#57 0x00007fe9753d931e clang::Sema::InstantiateFunctionDefinition(clang::SourceLocation, clang::FunctionDecl*, bool, bool, bool) (/lib64/libclang-cpp.so.18.1+0x17d931e)
#58 0x00007fe9753dc107 clang::Sema::PerformPendingInstantiations(bool) (/lib64/libclang-cpp.so.18.1+0x17dc107)
#59 0x00007fe9753d931e clang::Sema::InstantiateFunctionDefinition(clang::SourceLocation, clang::FunctionDecl*, bool, bool, bool) (/lib64/libclang-cpp.so.18.1+0x17d931e)
#60 0x00007fe9753dc107 clang::Sema::PerformPendingInstantiations(bool) (/lib64/libclang-cpp.so.18.1+0x17dc107)
#61 0x00007fe9753d931e clang::Sema::InstantiateFunctionDefinition(clang::SourceLocation, clang::FunctionDecl*, bool, bool, bool) (/lib64/libclang-cpp.so.18.1+0x17d931e)
#62 0x00007fe9753dc107 clang::Sema::PerformPendingInstantiations(bool) (/lib64/libclang-cpp.so.18.1+0x17dc107)
#63 0x00007fe974b0547f clang::Sema::ActOnEndOfTranslationUnitFragment(clang::Sema::TUFragmentKind) (/lib64/libclang-cpp.so.18.1+0xf0547f)
#64 0x00007fe974b05c09 clang::Sema::ActOnEndOfTranslationUnit() (/lib64/libclang-cpp.so.18.1+0xf05c09)
#65 0x00007fe97425114d clang::Parser::ParseTopLevelDecl(clang::OpaquePtr<clang::DeclGroupRef>&, clang::Sema::ModuleImportState&) (/lib64/libclang-cpp.so.18.1+0x65114d)
#66 0x00007fe97417442e clang::ParseAST(clang::Sema&, bool, bool) (/lib64/libclang-cpp.so.18.1+0x57442e)
#67 0x00007fe97641c896 clang::FrontendAction::Execute() (/lib64/libclang-cpp.so.18.1+0x281c896)
#68 0x00007fe9763945a0 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/lib64/libclang-cpp.so.18.1+0x27945a0)
#69 0x00007fe976499df5 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/lib64/libclang-cpp.so.18.1+0x2899df5)
#70 0x00000000004144c6 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/usr/bin/clang+++0x4144c6)
#71 0x0000000000411164 (/usr/bin/clang+++0x411164)
#72 0x00007fe975fc523d (/lib64/libclang-cpp.so.18.1+0x23c523d)
#73 0x00007fe96cf35e24 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/lib64/libLLVM.so.18.1+0x535e24)
#74 0x00007fe975fc4ca7 clang::driver::CC1Command::Execute(llvm::ArrayRef<std::optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const (/lib64/libclang-cpp.so.18.1+0x23c4ca7)
#75 0x00007fe975f8aa07 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&, bool) const (/lib64/libclang-cpp.so.18.1+0x238aa07)
#76 0x00007fe975f8ac67 clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&, bool) const (/lib64/libclang-cpp.so.18.1+0x238ac67)
#77 0x00007fe975fa90ce clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&) (/lib64/libclang-cpp.so.18.1+0x23a90ce)
#78 0x00000000004109bf clang_main(int, char**, llvm::ToolContext const&) (/usr/bin/clang+++0x4109bf)
#79 0x000000000041dcba main (/usr/bin/clang+++0x41dcba)
#80 0x00007fe96c43d088 __libc_start_call_main (/lib64/libc.so.6+0x2a088)
#81 0x00007fe96c43d14b __libc_start_main@GLIBC_2.2.5 (/lib64/libc.so.6+0x2a14b)
#82 0x000000000040d8a5 _start (/usr/bin/clang+++0x40d8a5)
clang++: error: clang frontend command failed with exit code 139 (use -v to see invocation)
clang version 18.1.1 (Fedora 18.1.1-1.fc40)
Target: x86_64-redhat-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
clang++: note: diagnostic msg:
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang++: note: diagnostic msg: /tmp/wm-actions-95e088.cpp
clang++: note: diagnostic msg: /tmp/wm-actions-95e088.sh
clang++: note: diagnostic msg:

********************
```

[logs.zip](https://github.com/user-attachments/files/15682137/logs.zip)

**Comment by DimitryAndric** [comment]
This is a duplicate of #85343, which was fixed in `llvmorg-19-init-05872-g`f5f3d5d6534. Unfortunately, this was never backported to the 18.x branch, so you may have to ask your package maintainer to backport it manually.

(I did so for FreeBSD in https://github.com/freebsd/freebsd-src/commit/49a6e426df84eff1ae54905a02f66910a6a177d3).

**Comment by DimitryAndric** [comment]
cc @tstellar (are you still Fedora/RH package maintainer?)
